### ISSUE TITLE
.

### ISSUE THREAD

**Issue by DigitalSolutionsCZ** [pglombardo/PasswordPusher#1054]

**Comment by pglombardo** [comment]
Hi @DigitalSolutionsCZ - I don't completely understand the question.  

You need to set the `DATABASE_URL` in the pwpush container.  There is [documentation here](https://github.com/pglombardo/PasswordPusher/blob/master/containers/docker/README.md#pwpush-mysql).

Let me know if this helps but if not, could you rephrase your question to be a bit more clear?

**Comment by DigitalSolutionsCZ** [comment]
Hello, yes i set environment DATABASE_URL from your documentation but my pwpush image log now is:
 

Password Pusher: migrating database to latest...

rake aborted!

ActiveRecord::DatabaseConnectionError: There is an issue connecting with your hostname: mysql.

Please check your database configuration and ensure there is a valid connection to your database.

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/mysql2_adapter.rb:51:in `rescue in new_client'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/mysql2_adapter.rb:43:in `new_client'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/mysql2_adapter.rb:23:in `mysql2_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `public_send'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `new_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:700:in `checkout_new_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:679:in `try_to_checkout_new_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:640:in `acquire_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:341:in `checkout'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:181:in `connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_handler.rb:211:in `retrieve_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_handling.rb:313:in `retrieve_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_handling.rb:280:in `connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/tasks/database_tasks.rb:262:in `migrate'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/railties/databases.rake:92:in `block (2 levels) in <main>'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli/exec.rb:58:in `load'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli/exec.rb:58:in `kernel_load'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli/exec.rb:23:in `run'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli.rb:486:in `exec'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli.rb:31:in `dispatch'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli.rb:25:in `start'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/exe/bundle:48:in `block in <top (required)>'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/exe/bundle:36:in `<top (required)>'

/usr/local/bin/bundle:25:in `load'

/usr/local/bin/bundle:25:in `<main>'

Caused by:

Mysql2::Error::ConnectionError: Unknown server host 'mysql' (-2)

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:97:in `connect'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:97:in `initialize'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/mysql2_adapter.rb:44:in `new'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/mysql2_adapter.rb:44:in `new_client'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/mysql2_adapter.rb:23:in `mysql2_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `public_send'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `new_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:700:in `checkout_new_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:679:in `try_to_checkout_new_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:640:in `acquire_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:341:in `checkout'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:181:in `connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/abstract/connection_handler.rb:211:in `retrieve_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_handling.rb:313:in `retrieve_connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/connection_handling.rb:280:in `connection'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/tasks/database_tasks.rb:262:in `migrate'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.4.3/lib/active_record/railties/databases.rake:92:in `block (2 levels) in <main>'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli/exec.rb:58:in `load'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli/exec.rb:58:in `kernel_load'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli/exec.rb:23:in `run'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli.rb:486:in `exec'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli.rb:31:in `dispatch'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/cli.rb:25:in `start'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/exe/bundle:48:in `block in <top (required)>'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'

/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bundler-2.3.25/exe/bundle:36:in `<top (required)>'

/usr/local/bin/bundle:25:in `load'

/usr/local/bin/bundle:25:in `<main>'

Tasks: TOP => db:migrate

(See full trace by running task with --trace)

**Comment by pglombardo** [comment]
`mysql` needs to be the hostname to your database server.  If you are using Docker compose files, the usual solution is to link:

````
links:
     - db:mysql
```

How are you running Password Pusher?  With Docker Compose files?

**Comment by DigitalSolutionsCZ** [comment]
With "links" is log pwpush container still same (because we use docker stack in docker swarm).
Yes and I check your example but not working for us.
With docker stack in one of multiple node in docker swarm.

**Comment by pglombardo** [comment]
The crux of your problem is that the container tries to do a DNS lookup of `mysql` but fails because that record doesn't exist.  Change `mysql` to the hostname of your MariaDB server.

**Comment by pglombardo** [comment]
Did you want to close this issue?  I hope it helped at least.  If I can help any further, let me know.
